### ISSUE TITLE
Sounds for parallelogram special case identification (equal side lengths, right angles, etc)

### ISSUE THREAD

**Issue by BLFiedler** [phetsims/quadrilateral#58]
From the Feb 2022 remote interviews it seems desirable for additional indications that "special cases" of the parallelogram are met, which result in other notable shapes, i.e., square, rectangle, and rhombus. While there may be additional visual cues to emphasize these moments, there may not be description space for this and a quick sonification may be more efficient.

We would like to sonify state changes between two (three if distinct when combined) shape properties when the shape is a parallelogram:
1. All side lengths are equal (rhombus)
2. All angles are right angles (rectangle)
3. Both (square)

We would like to create two to three sounds that are short in nature and only play when the conditions are met (not when leaving the state). They are ideally informatory, but do not give an indication of success or failure. These sounds may be playing at the same time as the "parallelogram maintenance" sound. They will like by overridden by the parallelogram success sound, but should not mask it in any case.

**Comment by Ashton-Morris** [comment]
[I have added two sounds for the above issue.](https://drive.google.com/drive/u/0/folders/1YNrthgUqZXTQw-gA8ch6XIwBZ_RhsIKW)

- all-angles-are-right-angles.mp3
- all-side-lengths-are-equal.mp3

I am assuming for "Both" they will both be triggered automatically when conditions are met so I did not export a 3rd sounds comprised of both sounds.

**Comment by jessegreenberg** [comment]
These sounds were added. They are being played when we detect a named shape of square, rectangle, or rhombus. Here is the code:
```js
        // Unique sounds indicating that all angles ore lengths are equal. We do this when we achieve the appropriate
        // shapes so that these sounds match the shape detection of the model. If we linked this to angle or length
        // values, we may encounter cases where a sound is played indicating that all angles are the same but
        // the model is also not a square/rectangle because of tolerance values in the calculations.
        if ( name === NamedQuadrilateral.RECTANGLE ) {

          // a unique sound indicating that all angles are the same
          this.allRightAnglesSoundClip.play();

        }
        else if ( name === NamedQuadrilateral.RHOMBUS ) {

          // a unique sound indicating all lengths are the same
          this.allLengthsEqualSoundClip.play();
        }
        else if ( name === NamedQuadrilateral.SQUARE ) {

          // both angles and lengths are all the same
          this.allRightAnglesSoundClip.play();
          this.allLengthsEqualSoundClip.play();
        }
```

As documented, I did this so that we don't use a different tolerance for playing these sounds than we do when detecting named shapes. This makes sure that the sounds match the right angle indicators, and match the detected shape in the model. If these sounds used their own angle comparisons they might play even when the shape is not actually a square/rectangle/rhombus because of tolerance intervals.

@BLFiedler @Ashton-Morris can you please review?

**Comment by BLFiedler** [comment]
Currently the "all right angles" sound plays when transitioning from a rectangle to a square. Let's only have the sound play when the All Right Angles condition happens the first time, not replay when a square is achieved. The sound should be tied to the "all right angles" and not the shape itself.

So, moving from a random parallelogram to a rectangle would play the "snare" all right angles sound. Moving from a rectangle to a square by shortening the sides would only play the "tom" all equal sides sound. 

This means they could be slightly offset if someone is trying to create a square by moving a vertex, but I think that is fine.

**Comment by BLFiedler** [comment]
@Ashton-Morris will create some variations of the Snare sound to dampen it, but retain the general feel and contrast to the tom sound.

**Comment by Ashton-Morris** [comment]
I have a few to test out. Uploading them now

- all-angles-are-right-angles-v2.mp3
- all-angles-are-right-angles-v4.mp3
- all-angles-are-right-angles-v3.mp3

**Comment by jessegreenberg** [comment]
I replaced it with v3 because that was my personal favorite! It is a little softer than the others and I think it fits in nicely with the other sounds. But @BLFiedler or @Ashton-Morris if you would like to hear the others in-sim let me know. v3 is now in master, assigning to @Ashton-Morris and @BLFiedler to review.

**Comment by jessegreenberg** [comment]
>  Let's only have the sound play when the All Right Angles condition happens the first time, not replay when a square is achieved. The sound should be tied to the "all right angles" and not the shape itself.

The reason it was tied to the shape was because I wanted the sound to play when it was actually a square in the model. I was worried if we used angle comparisons here they could differ from the angle comparisons done to detect shape. Same issue as https://github.com/phetsims/quadrilateral/issues/60#issuecomment-1044955319.

**Comment by BLFiedler** [comment]
Ah. Hm. Okay. Let me spend some time with the tolerances and we can revisit this. If it feels important, perhaps we can put in a hacky fix like forbidding the "all right angle sounds" when going directly from square to rectangle or something. However, I'd like to try to base all of this on the tolerances first if possible.

**Comment by jessegreenberg** [comment]
I think I have a different solution for this in the above commit that only plays the sounds when the condition is met (instead of every time the shape name changes), but the condition is still dependent on the shape name. @BLFiedler I think everything here is ready for review again.

**Comment by BLFiedler** [comment]
That solution should work fine for our current scaffolding. It might become a problem if we ever scaffold non-parallelograms. So far it feels alright!

One thing I have noticed is it is very hard to hear the Tom (all sides equal) when the Parallelogram Maintenance sound is playing. All-Right-Angles is pretty easily heard. Can we raise the relative volume of the Tom?

**Comment by jessegreenberg** [comment]
Sounds good. I tried to increase the output level of the "tom" sound but I didn't hear any difference even when increasing it 50%. But it does start to distort. Reducing the output of the parallelogram maintenance sounds instead to see how that goes.

The "maintenance" sounds are already turned way down, output level was 0.2. I reduced it to 0.1. @BLFiedler can you please review?

**Comment by BLFiedler** [comment]
I lose the Tom sound on smaller speakers. Bringing up at meeting, but pinging @Ashton-Morris to put on his radar to design around potentially?

**Comment by BLFiedler** [comment]
@Ashton-Morris mentioend being able to add some "harmonics" or similar that folks use on more bassy sounds so they are audible on a wider range of speakers. He'll create a variation or two to try out.

**Comment by Ashton-Morris** [comment]
> @Ashton-Morris mentioned being able to add some "harmonics" or similar that folks use on more bassy sounds so they are audible on a wider range of speakers. He'll create a variation or two to try out.

I have added some high end and harmonics too test out. [Please see file here.](https://drive.google.com/file/d/1UPbodaVqBXIpHjwWx6_GEPa_fVrQBEBg/view?usp=sharing)

**Comment by jessegreenberg** [comment]
The sound from https://github.com/phetsims/quadrilateral/issues/58#issuecomment-1068449868 has replaced the old tom sound. @BLFiedler back to you to review.

**Comment by BLFiedler** [comment]
I can hear it on smaller speakers for sure! Let's plan on discussing at sound meeting.

**Comment by BLFiedler** [comment]
To check w/ JG next week: Is the volume mixing still set to make up for the lack of audibility of the tom sound? Making sure some of the "blown-out"-ness of the sound is not due to anything like that rather than the sound itself.

**Comment by jessegreenberg** [comment]
@BLFiedler happy to discuss more details but you can hear the original sound file here: https://drive.google.com/file/d/1UPbodaVqBXIpHjwWx6_GEPa_fVrQBEBg/view

It sounds a little fuzzy to me too, if that is that what you mean by "blown-out"-ness.
